eng/scripts/Language-Settings.ps1

```diff
@@ -1,41 +1,52 @@
 $Language = "go"
 $packagePattern = "go.mod"
-# rewrite from ChangeLog-Operations.ps1 used in Get-ChangeLogEntriesFromContent for go uses vx.x.x as version number
-$RELEASE_TITLE_REGEX = "(?<releaseNoteTitle>^\#+\s+(?<version>v$([AzureEngSemanticVersion]::SEMVER_REGEX))(\s+(?<releaseStatus>\(.+\))))"
+$LanguageDisplayName = "go"
+$VERSION_LINE_REGEX = ".+\s*=\s*`".*v(?<version>$([AzureEngSemanticVersion]::SEMVER_REGEX))`""
 
-# rewrite from artifact-metadata-parsing.ps1 used in RetrievePackages for fetch go single module info
-function Get-go-PackageInfoFromPackageFile ($pkg, $workingDirectory)
+function Get-GoPackageId ($pkgDirectory) 
 {
-    $workFolder = $pkg.Directory
-    $releaseNotes = ""
+    $serviceDir = ""
+    $pkgId = ""
+    $sdkType = ""
 
-    if ($workFolder -match "sdk[\/|\\]((?<repopath>(?<service>.*?)[\/|\\])?(?<arm>arm)?(?<pkgname>.*))")
+    if ($pkgDirectory -match "sdk[\/|\\]((?<repopath>(?<service>.*?)[\/|\\])?(?<arm>arm)?(?<pkgname>.*))")
     {
         if ($matches["arm"])
         {
             $packageName = "arm" + $matches["pkgname"]
-            $rpName = $matches["service"]
-            $pkgId = "sdk/$rpName/$packageName"
+            $serviceDir = $matches["service"]
+            $pkgId = "sdk/$serviceDir/$packageName"
+            $sdkType = "mgmt"
         }
         else
         {
             $packageName = $matches["pkgname"]
+            $serviceDir = $packageName
             $pkgId = "sdk/$packageName"
+            $sdkType = "client"
         }
     }
+    return $pkgId, $serviceDir, $sdkType
+}
+
+# rewrite from artifact-metadata-parsing.ps1 used in RetrievePackages for fetch go single module info
+function Get-go-PackageInfoFromPackageFile ($pkg, $workingDirectory)
+{
+    $workFolder = $pkg.Directory
+    $releaseNotes = ""
 
-    $pkgVersion = Get-Version $workFolder
+    $packageProperties = (Get-AllPackageInfoFromRepo -pkgDirectory $workFolder)[0]
 
-    $changeLogLoc = @(Get-ChildItem -Path $workFolder -Recurse -Include "CHANGELOG.md")[0]
-    if ($changeLogLoc)
+    if ($packageProperties.ChangeLogPath -and $packageProperties.Version)
     {
-        $releaseNotes = Get-ChangeLogEntryAsString -ChangeLogLocation $changeLogLoc -VersionString v$pkgVersion
+        $releaseNotes = Get-ChangeLogEntryAsString -ChangeLogLocation $packageProperties.ChangeLogPath `
+        -VersionString $packageProperties.Version
```

Comment:
$packageProperties.Version may not include `v`, which will cause failure of changelog search.

Reply:
v's have been removed from the CHANGELOG.md files, https://github.com/Azure/azure-sdk-for-go/pull/15524